torchvision/transforms/v2/functional/_geometry.py

```diff
@@ -1775,36 +1733,20 @@ def elastic_image(
         dtype = torch.float32
 
     # We are aware that if input image dtype is uint8 and displacement is float64 then
-    # displacement will be casted to float32 and all computations will be done with float32
+    # displacement will be cast to float32 and all computations will be done with float32
     # We can fix this later if needed
 
-    expected_shape = (1,) + shape[-2:] + (2,)
+    expected_shape = (1, height, width, 2)
     if expected_shape != displacement.shape:
         raise ValueError(f"Argument displacement shape should be {expected_shape}, but given {displacement.shape}")
 
-    if ndim > 4:
-        image = image.reshape((-1,) + shape[-3:])
-        needs_unsquash = True
-    elif ndim == 3:
-        image = image.unsqueeze(0)
-        needs_unsquash = True
-    else:
-        needs_unsquash = False
-
     if displacement.dtype != dtype or displacement.device != device:
         displacement = displacement.to(dtype=dtype, device=device)
 
-    image_height, image_width = shape[-2:]
-    grid = _create_identity_grid((image_height, image_width), device=device, dtype=dtype).add_(displacement)
-    output = _apply_grid_transform(image, grid, interpolation.value, fill=fill)
-
-    if needs_unsquash:
-        output = output.reshape(shape)
-
-    if is_cpu_half:
-        output = output.to(torch.float16)
-
-    return output
+    grid = _create_identity_grid((height, width), device=device, dtype=dtype).add_(
+        displacement.to(dtype=dtype, device=device)
```

Comment:
Is it equivalent to cast to float16 **before** the call to `_apply_grid_transform`? It used to be done after.

Reply:
Also why is `to(device)` needed now?

Reply:
The float16 handling above is untouched and deals with the image. Here we are just aligning the `displacement` with the `grid`. This is the same as

https://github.com/pytorch/vision/blob/f96deba0f834d7b9566e8f3c1783d9e0a1c5b5af/torchvision/transforms/v2/functional/_geometry.py#L1794-L1795

but relying on the noop behavior of `.to()` to avoid the check on our side.